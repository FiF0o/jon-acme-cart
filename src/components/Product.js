import React from 'react'
import PropTypes from 'prop-types';

export const Product = ({ productCode, name, price, productImg }) => (
  <div className="mdc-card demo-basic-with-header">
    <div className="demo-card__primary">
      <h2 className="demo-card__title mdc-typography mdc-typography--headline6">{ name }</h2>
      <h3 className="demo-card__subtitle mdc-typography mdc-typography--subtitle2">{ productCode }</h3>
    </div>
    <div className="mdc-card__primary-action demo-card__primary-action" tabIndex="0">
      <div 
        className="mdc-card__media mdc-card__media--16-9 demo-card__media"
        style={{
          backgroundImage: `url(${productImg})`
        }}
        ></div>
      <div className="demo-card__secondary mdc-typography mdc-typography--body2">Â£{ price }</div>
    </div>
  </div>
)

Product.propTypes = {
  productCode: PropTypes.string.isRequired,
  productImg: PropTypes.string,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired
}