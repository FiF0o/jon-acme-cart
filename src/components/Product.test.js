import React from 'react'
import { mount } from 'enzyme'

import { Product } from './Product'

describe(`<Product /> component`, () => {
  it('should render correctly', () => {
    const props = {
      productCode: 'ABC',
      name: 'name',
      price: 1,
      productImg: '/some/link/to/img'
    }

    const wrapper = mount(
      <Product {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})