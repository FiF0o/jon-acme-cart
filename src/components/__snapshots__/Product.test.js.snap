// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> component should render correctly 1`] = `
<div
  className="mdc-card demo-basic-with-header"
>
  <div
    className="demo-card__primary"
  >
    <h2
      className="demo-card__title mdc-typography mdc-typography--headline6"
    >
      name
    </h2>
    <h3
      className="demo-card__subtitle mdc-typography mdc-typography--subtitle2"
    >
      ABC
    </h3>
  </div>
  <div
    className="mdc-card__primary-action demo-card__primary-action"
    tabIndex="0"
  >
    <div
      className="mdc-card__media mdc-card__media--16-9 demo-card__media"
      style={
        Object {
          "backgroundImage": "url(/some/link/to/img)",
        }
      }
    />
    <div
      className="demo-card__secondary mdc-typography mdc-typography--body2"
    >
      Â£
      1
    </div>
  </div>
</div>
`;
